src/payload/addr.rs

```diff
@@ -0,0 +1,711 @@
+            return Err(PrefixError::LenOverflow)
+        }
+        Ok(Self::new_unchecked(
+            Bits::from_v6(addr).clear_host(len), len, true
+        ))
+    }
+
+    /// Returns whether the prefix is for an IPv4 address.
+    pub fn is_v4(self) -> bool {
+        self.family_and_len & 0x80 == 0
+    }
+
+    /// Returns whether the prefix is for an IPv6 address.
+    pub fn is_v6(self) -> bool {
+        self.family_and_len & 0x80 == 0x80
+    }
+
+    /// Returns the IP address part of a prefix.
+    pub fn addr(self) -> IpAddr {
+        if self.is_v4() {
+            self.bits.into_v4().into()
+        }
+        else {
+            self.bits.into_v6().into()
+        }
+    }
+
+    /// Returns the length part of a prefix.
+    #[allow(clippy::len_without_is_empty)]
+    pub fn len(self) -> u8 {
+        self.family_and_len & 0x7F
+    }
+
+    /// Returns the smallest address of the prefix.
+    ///
+    /// This is the same as [`addr`][Self::addr].
+    pub fn min_addr(self) -> IpAddr {
+        self.addr()
+    }
+
+    /// Returns the largest address of the prefix.
+    pub fn max_addr(self) -> IpAddr {
+        let bits = self.bits.into_max(self.len());
+        if self.is_v4() {
+            bits.into_v4().into()
+        }
+        else {
+            bits.into_v6().into()
+        }
+    }
+
+    /// Returns whether the prefix `self` covers the prefix `other`.
+    pub fn covers(self, other: Self) -> bool {
+        // Differing families? Not covering.
+        if self.is_v4() != other.is_v4() {
+            return false
+        }
+
+        // If self is more specific than other, it can’t cover it.
+        if self.len() > other.len() {
+            return false
+        }
+
+        // If we have two host prefixes, they need to be identical.
+        // (This needs to be extra because the bit shifting below doesn’t
+        // work at least in the v6 case.)
+        if self.is_v4() {
+            if self.len() == 32 && other.len() == 32 {
+                return self == other
+            }
+        }
+        else if self.len() == 128 && other.len() == 128 {
+            return self == other
+        }
+
+        // other now needs to start with the same bits as self.
+        let left = self.bits.into_int() & !(u128::MAX >> self.len());
+        let right = other.bits.into_int() & !(u128::MAX >> self.len());
+        left == right
+    }
+}
+
+
+//--- From
+
+#[cfg(feature = "repository")]
+impl From<crate::repository::roa::FriendlyRoaIpAddress> for Prefix {
+    fn from(addr: crate::repository::roa::FriendlyRoaIpAddress) -> Self {
+        Prefix::new(
+            addr.address(), addr.address_length()
+        ).expect("ROA IP address with illegal prefix length")
+    }
+}
+
+#[cfg(feature = "repository")]
+impl From<Prefix> for crate::repository::resources::IpBlock {
+    fn from(src: Prefix) -> Self {
+        crate::repository::resources::Prefix::new(
+            src.addr(), src.len()
+        ).into()
+    }
```

Comment:
They actually don’t. `repository::resources::Prefix` is different. These will be merged, but this requires a bit of careful checking, since this type currently is not aware of its address family. I’d like to leave that for a separate PR.